modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -187,24 +194,95 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
    * Workaround for http://bugs.mysql.com/bug.php?id=65909
    */
   private void shutdownMySqlAbandonedConnectionCleanupThread() {
-
-    Class<?> classAbandonedConnectionCleanupThread;
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      Class<?> cleanupThreadsClass = findMySqlDriverClass();
+      if (cleanupThreadsClass == null) {
+        logger.warn("No AbandonedConnectionCleanupThread class found. Check if thread class is between "
+            + CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.toString());
+        throw new ClassNotFoundException(CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.get(0));
+      }
+      shutdownMySqlConnectionCleanupThreads(cleanupThreadsClass);
       try {
-        Method uncheckedShutdown = classAbandonedConnectionCleanupThread.getMethod("uncheckedShutdown");
-        uncheckedShutdown.invoke(null);
-      } catch (NoSuchMethodException e) {
-        Method checkedShutdown = classAbandonedConnectionCleanupThread.getMethod("shutdown");
-        checkedShutdown.invoke(null);
+        // The cleanup threads are fired from a single-thread ThreadPoolExecutor, which is created inside a
+        // lambda, which wraps the thread pool into a finalizable wrapper. This leads to retention in the
+        // artifact classloaders when several redeployments are performed.
+        cleanMySqlCleanupThreadsThreadFactory(cleanupThreadsClass);
+      } catch (NoSuchFieldException e) {
+        logger.warn("Error cleaning threadFactory from AbandonedConnectionCleanupThread executor service");
       }
     } catch (ClassNotFoundException | NoSuchMethodException | SecurityException | IllegalAccessException
         | IllegalArgumentException | InvocationTargetException e) {
       logger.warn("Unable to shutdown MySql's AbandonedConnectionCleanupThread", e);
     }
   }
 
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1148))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAt&open=AWjzLS16otIXnmsUYkAt)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAt